stripe/src/main/java/com/stripe/android/view/PaymentMethodsAdapter.kt

```diff
@@ -165,13 +158,7 @@ internal class PaymentMethodsAdapter constructor(
     @JvmSynthetic
     internal fun onPositionClicked(position: Int) {
         updateSelectedPaymentMethod(position)
-        CoroutineScope(workContext).launch {
-            delay(0)
-
-            withContext(Dispatchers.Main) {
-                listener?.onPaymentMethodClick(getPaymentMethodAtPosition(position))
-            }
-        }
```

Comment:
Why was this done this way?

Reply:
I can't remember now, but it wasn't actually doing anything because `delay(0)` is a no-op:


```
public suspend fun delay(timeMillis: Long) {
    if (timeMillis <= 0) return // don't delay
    return suspendCancellableCoroutine sc@ { cont: CancellableContinuation<Unit> ->
        // if timeMillis == Long.MAX_VALUE then just wait forever like awaitCancellation, don't schedule.
        if (timeMillis < Long.MAX_VALUE) {
            cont.context.delay.scheduleResumeAfterDelay(timeMillis, cont)
        }
    }
}
```